src/main/java/org/jboss/tm/ExtendedJBossXATerminator.java

```diff
@@ -91,8 +91,9 @@
      *                      set in the parameter. These constants are defined in javax.transaction.xa.XAResource interface
      * @return an array of Xids to recover
      * @throws XAException An error has occurred. Possible values are XAER_RMERR, XAER_RMFAIL, XAER_INVAL, and XAER_PROTO.
+     * @throws UnsupportedOperationException Operation is not supported in the context. This happens when Narayana runs on JTS.
      */
-    Xid[] getXidsToRecoverForParentNode(boolean recoverInFlight, final String parentNodeName, int recoveryFlags) throws XAException;
+    Xid[] getXidsToRecoverForParentNode(boolean recoverInFlight, final String parentNodeName, int recoveryFlags) throws XAException, UnsupportedOperationException;
```

Comment:
and I think it won't. the wildfly txn client uses some methods from interface in it's implementation of `XAImporter`

https://github.com/wildfly/wildfly-transaction-client/blob/master/src/main/java/org/wildfly/transaction/client/provider/jboss/JBossLocalTransactionProvider.java#L82

https://github.com/wildfly/wildfly-transaction-client/blob/master/src/main/java/org/wildfly/transaction/client/provider/jboss/JBossLocalTransactionProvider.java#L546